onnxscript/ir/_core.py

```diff
@@ -1187,18 +1189,18 @@ def __init__(
         self._version: int | None = version
         self._metadata: _metadata.MetadataStore | None = None
         self._metadata_props: dict[str, str] | None = metadata_props
-        self._graph: Graph | Function | None = graph
+        # _graph is set by graph.append
+        self._graph: Graph | None = None
+        # Add the node to the graph if graph is specified
+        if graph is not None:
+            graph.append(self)
```

Comment:
What happens if you append a node into a graph != node.graph? I tend to reuse nodes when I tweak graphs.

Reply:
It will raise. You must set `node.graph=None` before adding it into a new graph to make the intention clear